docs/proposals/use-cascading-deletion/README.md

```diff
@@ -0,0 +1,137 @@
+- TBD
+
+approvers:
+- "@robot"
+- TBD
+
+creation-date: 2024-07-01
+
+# Use Cascading Deletion in Karmada
+
+## Summary
+<!--
+一种级联删除机制, 类似于[级联删除](https://kubernetes.io/docs/tasks/administer-cluster/use-cascading-deletion/)
+-->
+A cascading deletion mechanism, similar to [cascade deletion](https://kubernetes.io/docs/tasks/administer-cluster/use-cascading-deletion/)
+
+## Motivation
+<!--
+默认情况下, 当用户删除Karmada控制面的资源模版之后,成员集群的资源也会被删除。但是仍然在某些场景下,用户希望仅仅删除控制面的资源,而保留成员集群资源;亦或是删除资源模版动作之后立即返回
+-->
+
+By default, when the user deletes the resource template of the Karmada control plane, the resources of the member cluster will also be deleted. However,in some scenarios,users want to delete only the control plane resources and retain the member cluster resources; or they want to return immediately after deleting the resource template action.
+
+### Goals
+<!--
+- 提供只删除控制面资源模板,而成员集群资源不删除,同时清理Karmada附加给成员集群资源的labels/annotations的能力
+- 提供控制面资源模版删除之后,不阻塞等待成员集群资源删除的能力
+- 其他直接产生work对象的资源如namespace(ns-controller开启)/cronfederatedhpa/federaredhpa/federatedresourcequota,需要有和资源模板相同的能力
+- 提供karmadactl命令行,可以执行级联删除策略, 如 `karmadactl delete deployment --cascade=orphan`
+-->
+
+- Provides the ability to delete only the control plane resource templates without deleting the member cluster resources, and at the same time clean up the labels/annotations attached to the member cluster resources by Karmada.
+- Provides the ability to not block waiting for member cluster resource deletion after the control plane resource template is deleted.
+- Other resources that directly generate work objects, such as namespace (ns-controller is turned on)/cronfederatedhpa/federaredhpa/federatedresourcequota, need to have the same capabilities as resource templates.
+- Provides karmadactl command line to execute cascade deletion strategy, such as `karmadactl delete deployment --cascade=orphan`
+
+### Non-Goals
+<!--
+- 不考虑不同的成员集群需要有不同的删除策略的情况
+- 其他删除策略, 比如保留work对象
+-->
+- Does not consider the situation that different member clusters need different deletion strategies
+- Other deletion strategies, such as retaining work objects
+
+## Proposal
+
+### User Stories (Optional)
+
+#### Story 1
+<!--
+作为管理员，我希望在工作负载迁移到Karmada期间有一个回滚机制，以便在出现任何意外情况时可以恢复到迁移前的状态。
+-->
+As an administrator, I would like to have a rollback mechanism during workload migration to Karmada so that in case of any unexpected situation, I can revert to the pre-migration state.
+
+#### Story 2
+<!--
+作为管理员, 我希望可以有一种手段可以将某个测试集群从Karmada控制面移除,同时测试集群上面的workload不受影响
+-->
+As an administrator, I hope there is a way to remove a test cluster from the Karmada control plane without affecting the workload on the test cluster.
+
+#### Story 3
+<!--
+作为管理员,为了降低接口耗时,我希望在删除控制面资源模板之后可以立即返回,不阻塞等待直到成员集群资源删除。
+-->
+As an administrator, in order to reduce the interface time-consuming, I hope that I can return immediately after deleting the control plane resource template without blocking and waiting until the member cluster resources are deleted.
```

Comment:
I have some confusion about this point, what exactly is meant by "interface time-consuming"? How can the reduction in consumption of the current plan be measured?